Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -249,46 +241,48 @@ public class ResourcesProjectMapper: ProjectMapping { // swiftlint:disable:this
             ), the bundle for classes within this module can be used directly.
             static let module = Bundle(for: BundleFinder.self)
             }
-
-            // MARK: - Objective-C Bundle Accessor
-
-            @objc
-            public class \(target.productName.toValidSwiftIdentifier())Resources: NSObject {
-            @objc public class var bundle: Bundle {
-                return .module
-            }
-            }
             // swiftlint:enable all
             // swiftformat:enable all
 
             """
         }
     }
 
-    static func objcHeaderFileContent(targetName: String, bundleName _: String, target _: Target) -> String {
+    static func objcHeaderFileContent(
+        targetName: String,
+
+        bundleName _: String,
+        target _: Target,
+        projectName: String
+    ) -> String {
         return """
         #import <Foundation/Foundation.h>
 
         #if __cplusplus
         extern "C" {
         #endif
 
-        NSBundle* \(targetName)_SWIFTPM_MODULE_BUNDLE(void);
+        NSBundle* \(projectName)_\(targetName)_SWIFTPM_MODULE_BUNDLE(void);
```

Comment:
SPM adds the Package name in addition to the target name (I'm guessing to reduce chances for name collision)
[SPM](https://github.com/apple/swift-package-manager/blob/26b4ed010e7ec211f0aad7527961f83508ea4771/Sources/Build/BuildDescription/ClangTargetBuildDescription.swift#L441)

[My Example](https://github.com/danibachar/SPM-Bundle-Accessor/blob/main/OBJC/Sources/ObjcTarget/Objc.m#L20)